src/main/java/com/arpnetworking/metrics/mad/parsers/JsonToRecordParser.java

```diff
@@ -430,6 +420,29 @@ private static void putVariablesVersion2fSteno(
         }
     }
 
+    private static void putVariablesVersion2g(
+            final Map<String, Version2g.Element> elements,
+            final MetricType metricKind,
+            final ImmutableMap.Builder<String, Metric> variables) {
+
+        for (final Map.Entry<String, Version2g.Element> entry : elements.entrySet()) {
+            final Version2g.Element element = entry.getValue();
+            final List<Quantity> quantities = Lists.newArrayList(
```

Comment:
Agreed, get rid of the Lists.newArrayList wrapping call.  The streams collector will create that list for you.